Docu (#3455)
ElektraInitiative/libelektra: src/bindings/jna/README.md

# JNA

A Java binding using JNA. This binding requires Elektra to be installed on
the system to work.

## Usage

To use the bindings in a Java project, we have to include the jar file
libelektra-\$VERSION.jar in the project. The version number is the same
one as used for Elektra. This jar is created upon build of Elektra if
you enable the jna bindings, e.g., with `cmake -DBINDINGS=jna`, see also
[COMPILE](/doc/COMPILE.md#bindings). Internally, `mvn` will be used to
actually compile the plugin.

> Please note that the [jni plugin](/src/plugins/jni) is a different thing. We

Reviewer: ```suggestion
Please note that the [jni plugin](/src/plugins/jni) serves a different purpose. We
```

libelektra.jar), e.g.:

```sh
export CLASSPATH=".:/usr/share/java/libelektra4j.jar:/usr/share/java/jna.jar"
```

> Or, if you want to use Elektra from cmake's build directory, use

Author: ```suggestion
```

> `

Reviewer: Did something go wrong here or is this intentional?

Author: Yes, this is an error, initially I wanted to recommend to use `scripts/dev/run_env` but it does not work so nice (installing with mvn seems to be the better option).

> `

Reviewer: ```suggestion
```

    If you copied the jna.jar, junit.jar and hamcrest-core.jar directly to the
    jna directory, the correct path would be `./jna.jar:./junit.jar:./hamcrest-core.jar`

>     (separated by : on mac and Linux, by ; on windows), otherwise specify the

Reviewer: ```suggestion
    (separated by : on Mac and Linux, by ; on Windows), otherwise specify the
```
